src/client.ts

```diff
@@ -141,6 +141,83 @@ export default class Client {
   clearCache() {
 
     this.cache.clear();
+    this.cacheDependencies = new Map();
+
+  }
+
+  /**
+   * Caches a State object
+   *
+   * This function will also emit 'update' events to resources, and store all
+   * embedded states.
+   */
+  cacheState(state: State) {
+
+    this.cache.store(state);
+    const resource = this.resources.get(state.uri);
+    if (resource) {
+      // We have a resource for this object, notify it as well.
+      resource.emit('update', state);
+    }
+
+    for(const embeddedState of state.getEmbedded()) {
+      // Recursion. MADNESS
+      this.cacheState(embeddedState);
+    }
+
+  }
+
+  /**
+   * cacheDependencies contains all cache relationships between
+   * resources.
+   *
+   * This lets a user (for example) let a resource automatically
+   * expire, if another one expires.
+   *
+   * A server can populate this list using the `inv-by' link.
+   */
+  public cacheDependencies: Map<string, Set<string>> = new Map();
+
+  /**
+   * Helper function for clearing the cache for a resource.
+   *
+   * This function will also emit the 'stale' event for resources that have
+   * subscribers, and handle any dependent resource caches.
+   *
+   * If any resources are specified in deletedUris, those will not
+   * receive 'stale' events, but 'delete' events instead.
+   */
+  clearResourceCache(staleUris: string[], deletedUris: string[]) {
```

Comment:
Are there tests for this stuff?

Reply:
Yes, the functionality was covered with tests and everything still has coverage and passes.